compiler/modules/nback/build.bal

```diff
@@ -916,41 +916,171 @@ function buildBitwiseBinary(llvm:Builder builder, Scaffold scaffold, bir:IntBitw
 }
 
 function buildCompare(llvm:Builder builder, Scaffold scaffold, bir:CompareInsn insn) returns BuildError? {
-    match insn.orderType {
+    var [lhsRepr, lhsValue] = check buildReprValue(builder, scaffold, insn.operands[0]);
+    var [rhsRepr, rhsValue] = check buildReprValue(builder, scaffold, insn.operands[1]);
+    bir:Register result = insn.result;
+
+    match [lhsRepr.base, rhsRepr.base] {
+        [BASE_REPR_TAGGED, BASE_REPR_INT] => {
+            buildCompareTaggedInt(builder, scaffold, buildIntCompareOp(insn.op), lhsValue, rhsValue, "lhs", result);
+        }
+        [BASE_REPR_INT, BASE_REPR_TAGGED] => {
+            buildCompareTaggedInt(builder, scaffold, buildIntCompareOp(insn.op), lhsValue, rhsValue, "rhs", result);
+        }
+        [BASE_REPR_TAGGED, BASE_REPR_FLOAT] => {
+            buildCompareTaggedFloat(builder, scaffold, buildFloatCompareOp(insn.op), lhsValue, rhsValue, "lhs", result);
```

Comment:
Instead of lhs/rhs, just put the op the other way round  (i.e. `<` goes to `>`, `<=` goes to `>=` and vice-versa).

Reply:
Fixed